### PR TITLE
feat: karmadactl init: grant clusterrole admin with karamda resource permission

### PR DESCRIPTION
**Pull request by zhy76**
**What type of PR is this?**

**What this PR does / why we need it**:
feat: karmadactl init: grant clusterrole admin with karamda resource permission
**Which issue(s) this PR fixes**:
Part of #3916 

**Special notes for your reviewer**:
**Does this PR introduce a user-facing change?**:

```release-note
karmadactl init: grant clusterrole admin with karamda resource permission
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
/cc @lonelyCZ @prodanlabs 
Can you help take a look?

File: pkg/karmadactl/cmdinit/karmada/rbac.go [hunk 1]
```diff
@@ -123,3 +124,119 @@ func grantAccessPermissionToAgent(clientSet kubernetes.Interface) error {
...
+		{
+			APIGroups: []string{"autoscaling.karmada.io"},
+			Resources: []string{
+				"cronfederatedhpas",
+				"cronfederatedhpas/status",
+				"federatedhpas",
+				"federatedhpas/status",
+			},
+			Verbs: []string{
+				"get",
```
Comment (jwcesign) [inline pkg/karmadactl/cmdinit/karmada/rbac.go:151]: Can we use `*`?

Comment (zhy76) [reply pkg/karmadactl/cmdinit/karmada/rbac.go:151]: The https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources says:
>Caution: Using wildcards in resource and verb entries could result in overly permissive access being granted to sensitive resources. For instance, if a new resource type is added, or a new subresource is added, or a new custom verb is checked, the wildcard entry automatically grants access, which may be undesirable. The [principle of least privilege](https://kubernetes.io/docs/concepts/security/rbac-good-practices/#least-privilege) should be employed, using specific resources and verbs to ensure only the permissions required for the workload to function correctly are applied.

I wonder if there are other hidden trouble with wildcards :)

Comment (jwcesign) [reply pkg/karmadactl/cmdinit/karmada/rbac.go:151]: I believe it is acceptable to use wildcards in specific API groups(like, autoscaling.karmada.io).

cc @XiShanYongYe-Chang

Comment (XiShanYongYe-Chang) [reply pkg/karmadactl/cmdinit/karmada/rbac.go:151]: Hi @zhy76, We actually added two clusterRoles: `karmada-edit` and `karmada-view`. These two clusterRoles have separate read and write permissions, so they need to be written separately.

Comment (zhy76) [reply pkg/karmadactl/cmdinit/karmada/rbac.go:151]: Oh,thanks,I will fix it.

**Comment by lonelyCZ** [comment]
> How can I test locally? I do like this:
> 
> ```shell
> make karmadactl
> cp ./karmadactl /usr/local/bin/karmadactl
> curl -L https://github.com/karmada-io/karmada/releases/download/v1.7.0-alpha.3/crds.tar.gz -o crds.tar.gz
> karmadactl init --kube-image-mirror-country=cn --crds ./crds.tar.gz
> ```
> 
> but still has some problem: image

It seems lack of git tag info locally. You can use `git tag <version>` to add a new tag and then build it.

**Comment by zhy76** [comment]
> > How can I test locally? I do like this:
> > ```shell
> > make karmadactl
> > cp ./karmadactl /usr/local/bin/karmadactl
> > curl -L https://github.com/karmada-io/karmada/releases/download/v1.7.0-alpha.3/crds.tar.gz -o crds.tar.gz
> > karmadactl init --kube-image-mirror-country=cn --crds ./crds.tar.gz
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > but still has some problem: image
> 
> It seems lack of git tag info locally. You can use `git tag <version>` to add a new tag and then build it.

Thanks~

**Comment by zhy76** [comment]
I have tested on my side:
```shell
git tag v1.7.0-alpha.3-60-g749f3843
hack/cli-testing-environment.sh
export KUBECONFIG="/root/karmada/karmada-apiserver.config”
```
image
karmada-edit:
image
karmada-view:
image
edit:
image
view:
image
admin:
image

**Comment by XiShanYongYe-Chang** [comment]
Hi @zhy76, is it ready to review?

**Comment by zhy76** [comment]
> Hi @zhy76, is it ready to review?你好，准备好审核了吗？

Yes

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm
/cc @lonelyCZ

**Comment by Vacant2333** [comment]
if karmadactl deinit need to delete these rbac resources?

**Comment by XiShanYongYe-Chang** [comment]
> if karmadactl deinit need to delete these rbac resources?

Good question. My understanding is that when calling `deinit`, it will be accompanied by the removal of the Karmada control plane, and these resources will also disappear. Ask @zhy76  to help confirm this again.

**Comment by zhy76** [comment]
> > if karmadactl deinit need to delete these rbac resources?
> 
> Good question. My understanding is that when calling `deinit`, it will be accompanied by the removal of the Karmada control plane, and these resources will also disappear. Ask @zhy76 to help confirm this again.

Yes, what I understand is that the karmada control plane saves the clusterrole in karmada's etcd, and after doing `deinit`, karmada's etcd will be deleted, which will delete the clusterrole.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**